spec/services/coffee_shop_service_spec.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Services::CoffeeShopService, method: :call do
+  let(:user_coords) { { x: 47.6, y: -122.4 } }
+  let(:coffee_shops) do
+    [
+      { name: 'Starbucks Seattle', x: 47.61, y: -122.33 },
+      { name: 'Starbucks SF', x: 37.77, y: -122.42 },
+      { name: 'Starbucks Seattle2', x: 47.63, y: -122.4 }
+    ]
+  end
+  let(:instance) { Services::CoffeeShopService.new(user_coords, coffee_shops) }
+  subject(:coffe_shop_service) { instance.call }
+
+  it 'responds with an Array of closest coffee shops' do
+    expected_coffee_shops_array =
+      [
+        { name: 'Starbucks Seattle2', x: 47.63, y: -122.4, distance: 0.03 },
+        { name: 'Starbucks Seattle', x: 47.61, y: -122.33, distance: 0.0707 },
+        { name: 'Starbucks SF', x: 37.77, y: -122.42, distance: 9.83 }
+      ]
+
+    expect(coffe_shop_service).to eq expected_coffee_shops_array
```

Comment:
coffe_shop_service sounds like it's referring to a CoffeeShopService instance, but then i see that it is actually defined as instance.call. 

This reminded me of a couple of quotes:
- naming is hard(TM)
- we write code for people, not machines
- write code as if the person maintaining your code is a serial killer and knows where you live 

😆